.github/workflows/libssl.yaml

```diff
@@ -24,15 +24,16 @@ jobs:
           - stable
           - beta
           - nightly
-        os: [ubuntu-latest]
+        # TODO(XXX): consider replacing ubuntu-24.04 w/ ubuntu-latest when appropriate
+        os: [ubuntu-24.04, ubuntu-22.04]
     steps:
       - name: Checkout sources
         uses: actions/checkout@v4
         with:
           persist-credentials: false
 
       - name: Install build dependencies
-        run: sudo apt-get update && sudo apt-get install -y openssl libssl3 libssl-dev lld
+        run: sudo apt-get update && sudo apt-get autopurge -y needrestart && sudo apt-get install -y openssl libssl3 libssl-dev lld
```

Comment:
FYI, GH have fixed their base images meaning it's no longer needed to purge needrestart.

Reply:
Oh wonderful, thanks for mentioning. I will back out this part of the diff.

Reply:
Removed.